UI/Windows/OptionsWindow/OptionsWindowHotkeys.cs

```diff
@@ -0,0 +1,172 @@
+﻿using SPCode.UI.Components;
+using SPCode.Interop;
+using System.IO;
+using System.Xml;
+using System;
+using SPCode.Utils;
+using System.Windows.Threading;
+using System.Windows.Input;
+using System.Linq;
+using System.Windows;
+using System.Windows.Media;
+using System.Windows.Controls;
+using System.Diagnostics;
+
+namespace SPCode.UI.Windows
+{
+    public partial class OptionsWindow
+    {
+        private readonly DispatcherTimer SaveHotkeyTimer;
+        private HotkeyEditorControl _ctrl;
+        private Hotkey _currentControlHotkey;
+        private FontStyle _currentFontStyle;
+
+        private void Hotkey_PreviewKeyDown(object sender, KeyEventArgs e)
+        {
+            var key = e.Key;
+
+            if (key == Key.System)
+            {
+                key = e.SystemKey;
+            }
+
+            if (key == Key.LeftCtrl ||
+                key == Key.RightCtrl ||
+                key == Key.LeftAlt ||
+                key == Key.RightAlt ||
+                key == Key.LeftShift ||
+                key == Key.RightShift ||
+                key == Key.LWin ||
+                key == Key.RWin ||
+                key == Key.Clear ||
+                key == Key.OemClear ||
+                key == Key.Apps)
+            {
+                return;
+            }
```

Comment:
Same as above

Reply:
Yep I might just work on that snippet to avoid repetition

Reply:
Yes you can create a small function and just call it.